internal/command/deploy/machines.go

```diff
@@ -624,11 +624,6 @@ func (md *machineDeployment) resolveUpdatedMachineConfig(origMachineRaw *api.Mac
 
 	if origMachineRaw.Config.Mounts != nil {
 		launchInput.Config.Mounts = origMachineRaw.Config.Mounts
-	} else if md.appConfig.Mounts != nil {
```

Comment:
I think this will work? resolveUpdatedMachineConfig() is used in the regular deployment as well, but those should all hit the first branch of the if statement (and I'm pretty certain we validate all the machines have mounts before getting to this part).

I'd recommend making some machines with volumes, deploying, adding a release command to the fly.toml and ensuring it still works. This script might be useful for prepping the volumes and machines:

```bash
#!/usr/bin/env bash
set -eo pipefail
set -x
FLY_API_BASE=https://api.machines.dev
app=tvd-test-apps-v2
volname=tvd_appsv2_test_vol
tag=test-machine-$(date '+%Y-%m-%dT%H%M%S')
count=1
for reg in $(fly platform regions --json | jq -r '.[].Code' | grep -v fra | grep -v maa | shuf -n ${count}) ; do
    vid=$(fly vol create -r ${reg} --size 1 --json ${volname} | jq -r '.id')
    curl -vs --fail -X POST "${FLY_API_BASE}/v1/apps/${app}/machines" \
    -H "Authorization: Bearer ${FLY_API_TOKEN}" \
    -H "content-type: application/json" \
    --data-binary '{
    "region": "'${reg}'",
    "config": {
        "image": "nginx",
        "guest": {
            "cpu_kind": "shared",
            "cpus": 1,
            "memory_mb": 512
        },
        "services": [{
            "protocol": "tcp",
            "internal_port": 80,
            "concurrency": {
                "type": "requests",
                "soft_limit": 2000,
                "hard_limit": 2500
            },
            "ports": [{
                "port": 443,
                "handlers": ["tls", "http"]
            }]
        }],
        "checks": {
            "vm": {
                "type": "http",
                "port": 80,
                "method": "GET",
                "path": "/",
                "interval": "15s",
                "timeout": "10s"
            }
        },
        "mounts": [{
            "volume": "'${vid}'",
            "path": "/testdata"
        }]
    }
}'
done

```

Reply:
Hi tvd,
 So I made a new fly app using the hello-fly template. I used the script above to create a machine with an attached volume. When i run fly deploy, it asks me to confirm converting the existing machine to the fly apps platform (if i say no it actually crashes flyctl, but that's gonna be a separate issue when I get around to it). If i say yes, the app deploys successfully. The release command does seem to run as well. If I try running with `echo 'hi'` as the command, it runs successfully, but if I use a command that will purposely fail (running `exit 1` as an example), the deployment fails. The machines are also correctly mounted to the volumes.

Here's the app config I used:
[hello-fly.zip](https://github.com/superfly/flyctl/files/10841884/hello-fly.zip)